### PR TITLE
HPCC-15608 Method Level Authentication

### PR DESCRIPTION
**Pull request by gaojx**
- Add ESDL attribute: auth_feature
- In Dynamic ESDL, the auth manager can use this attribute to auth a method

Signed-off-by: Jiafu Gao Jiafu.Gao@lexisnexis.com

### CODE REVIEW

**Comment by gaojx** [comment]
@rpastrana, @afishbeck please code review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,11 +69,11 @@
 
 project(hpccsystems-platform)
 
-if(UNIX AND NOT APPLE)
```
Comment (rpastrana) [inline CMakeLists.txt]: this cmake check shouldn't be removed.

File: esp/esdllib/esdl_transformer2.cpp [hunk 2]
```diff
@@ -1515,6 +1515,18 @@ int Esdl2Transformer::process(IEspContext &ctx, EsdlProcessMode mode, const char
...
         throw MakeStringException(-1, "ESDL - method '%s::%s'not found", service, method);
 
+    // check the auth_feature
+    if (mode==EsdlRequestMode)
+    {
+        const char* auth_feature = mi->queryMetaData("auth_feature");
+        if (auth_feature && *auth_feature)
+        {
+            SecAccessFlags access;
+            if (!ctx.authorizeFeature(auth_feature,access) || access!=SecAccess_Full)
```
Comment (rpastrana) [inline esp/esdllib/esdl_transformer2.cpp]: the auth_feature level should be configurable, here it's always assumed "FULL", even if SCApps will always set to full.
auth_feature("featurename:level");

**Review by rpastrana — CHANGES_REQUESTED** [review]
A couple of changes required.

**Comment by rpastrana** [comment]
@gaojx one more change needed.

**Comment by gaojx** [comment]
The access level is implemented:  The commit is: 9f1b5fa3df3b16a1ce21e789cae32e2bce593cb1. However it does not show up in github somehow.

**Comment by gaojx** [comment]
Now it shows up. Please code review again: @rpastrana

**Comment by rpastrana** [comment]
@gaojx looks fine.

**Comment by rpastrana** [comment]
@gaojx when you're ready, let Richard know this PR is ready to be merged.

**Comment by gaojx** [comment]
@richardkchapman this PR is ready for merging.

**Pull request merged**